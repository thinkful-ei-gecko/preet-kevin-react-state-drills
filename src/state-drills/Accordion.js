import React from 'react';

class Accordion extends React.Component {
  static defaultProps = {
    sections: []
  }
  state = {
    currentIndex: null
  }
  handleClick = index => {
    this.setState({ currentIndex: index });
  }
  render() {
    return (
      <ul>
        {
          this.props.sections.map((section, index) => {
            return (
              <li key={index}>
                <button onClick={() => this.handleClick(index)}>{section.title}</button>
                {this.state.currentIndex === index && <p>{section.content}</p>}
              </li>
            )
          })
        }
      </ul>
    )
  }
}

export default Accordion;

