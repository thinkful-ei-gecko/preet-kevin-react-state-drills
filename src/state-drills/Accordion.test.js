import React from 'react';
//import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]


describe('<Accordion />', () => {
  test('renders empty given no props', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  test('renders no section as deafult', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  
})