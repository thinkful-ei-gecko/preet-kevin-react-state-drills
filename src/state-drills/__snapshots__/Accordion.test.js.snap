// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders content from a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`<Accordion /> renders content from last section when multiple sections have been clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`<Accordion /> renders empty given no props 1`] = `<ul />`;

exports[`<Accordion /> renders no section by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
